### ISSUE TITLE
Possible optimization?

### ISSUE THREAD

**Issue by gdetari** [wolfgarbe/SymSpell#140]
In the following function:

```
private HashSet<string> Edits(string word, int editDistance, HashSet<string> deleteWords)
    {
        editDistance++;
        if (word.Length > 1)
        {
            for (int i = 0; i < word.Length; i++)
            {
                string delete = word.Remove(i, 1);
                if (deleteWords.Add(delete))
                {
                    //recursion, if maximum edit distance not yet reached
                    if (editDistance < maxDictionaryEditDistance) Edits(delete, editDistance, deleteWords);
                }
            }
        }
        return deleteWords;
    }
```

Would it make sense to only add to `deleteWords` if `editDistance < maxDictionaryEditDistance` is true? That is quite a difference in loading dictionary performance. It does change the functionality, but isn't that the correct approach not to add the word anyhow?

**Comment by wolfgarbe** [comment]
I think we already do exactly this. Edits is a recursive method. 
Only in the first iteration we unconditionally calculate and add deletes. 
That is because we always have maxDictionaryEditDistance>=1.
But the second and following iterations of Edits are done only if editDistance < maxDictionaryEditDistance.

**Comment by gdetari** [comment]
Yes, what I meant, that when editDistance == maxDictionaryEditDistance, we do add the delete word, but not call the recursion anymore. I was wondering if we need to add the delete word at all at this case, but now that I pondered on it more, I think it is intentional, right? That case should still be included, just not going anymore deeper with the recursion.
